### PR TITLE
time utility functions added

### PR DESCRIPTION
**Pull request by shahzadhaider1**
# Time Utilities Package

Added a comprehensive collection of time utility functions to simplify common date/time operations in Go. These utilities provide easy-to-use functions for:

- Date/time conversions and formatting
- Business day calculations
- Duration calculations and human-readable formatting
- Timezone conversions
- Age calculations
- Time comparisons and validations
- Calendar utilities (week numbers, month/day names)

All functions are thoroughly tested and documented with examples.

### Checklists:
* [x] Tests passing

#### If added a new utility function or updated existing function logic:
* [x] Updated [EXAMPLES.md](/EXAMPLES.md)
* [x] Updated [readME.md](/readME.md)

## Summary by CodeRabbit

- **New Features**
	- Added comprehensive time utility functions to the library.
	- Introduced 20 new time-related functions covering:
		- Date and time manipulation
		- Time zone conversions
		- Duration calculations
		- Date comparisons and formatting

- **Documentation**
	- Updated `EXAMPLES.md` with detailed function descriptions and code snippets.
	- Updated `readME.md` with a new Time Utilities section.

### CODE REVIEW

File: time/time.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+// EndOfDay returns the end of the day for the given time
+func EndOfDay(t time.Time) time.Time {
+	return time.Date(t.Year(), t.Month(), t.Day(), 23, 59, 59, 999999999, t.Location())
+}
+
+// AddBusinessDays adds the given number of business days to the given time
+func AddBusinessDays(t time.Time, days int) time.Time {
+	for days > 0 {
+		t = t.AddDate(0, 0, 1)
+		if t.Weekday() != time.Saturday && t.Weekday() != time.Sunday {
```
[the remark this answers is not in the document]
Comment (busi-reddy-karnati) [reply time/time.go:22]: @shahzadhaider1 Is there a reason why you skipped tests?
Let me know If I should add them.

[the remark this answers is not in the document]
Comment (shahzadhaider1) [reply time/time.go:22]: This is a Draft PR. I'll push the complete code shortly.

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]
Please resolve the conflicts

**Comment by shahzadhaider1** [comment]
Rebase done @kashifkhan0771 BTW multiple people updating the same Examples file is always gonna be a problem while merging PRs.

**Review by kashifkhan0771 — APPROVED** [review]
LGTM! I agree that updating `EXAMPLES.md` could lead to conflicts if multiple PRs are open simultaneously. :( That’s the only downside I can see, but I think the benefits outweigh the drawbacks.

**Comment by kashifkhan0771** [comment]
Well, both `time` and `fake` has index 13 in ReadMe now :)
When you get a chance, please fix it. Thank you!

**Comment by shahzadhaider1** [comment]
> Well, both `time` and `fake` has index 13 in ReadMe now :) When you get a chance, please fix it. Thank you!

well, here is another effect of multiple people editing same file (in reality, I hate writing documentation)

**Pull request merged**